### ISSUE TITLE
Ban does not detect after 'new' Keyword.

### ISSUE THREAD

**Issue by marvinside** [palantir/tslint#4216]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.2.0-dev.20181010
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
const a:Error = new Error("ABC"); 	// fine
const b:Error = Error("ABC"); 		// TSLint shows Error as Expected
```

with `tslint.json` configuration:

```json
"ban": [
	true,
	{"name": "Error", "message": "Use Meteor.Error instead."}
],
```

#### Actual behavior
Only the second Line is marked as an Error

#### Expected behavior
Both Lines (with and without new Keyword) should trigger an Error.

### Background Information
We use meteor to develop. Meteor provides the Class Meteor.Error (which extends the Normal Error).
This allows to include more details into the Error, but usually just Error is used.
By banning the Normal Error, developers should use the Meteor.Error and provide better Error messages.

**Comment by stnwk** [comment]
I would also love to have a solution to this problem. Anyone got any ideas?

**Comment by JoshuaKGoldberg** [comment]
Related: #2791

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
